src/requests.ts

```diff
@@ -540,20 +540,27 @@ export const ApiHeaderJson: RequestHeaderProducer<{}, "Content-Type"> = () => ({
  *
  * @param status  The response status handled by this decoder
  * @param type    The response type corresponding to the status
+ * @param processor A function that takes the object corresponding to the json response as input
+ * and returns the processed object (usefull when you want alterate the json body received)
  */
 export function ioResponseDecoder<
   S extends number,
   R,
   O = R,
   H extends string = never
->(status: S, type: t.Type<R, O>): ResponseDecoder<IResponseType<S, R, H>> {
+>(
+  status: S,
+  type: t.Type<R, O>,
+  // tslint:disable-next-line: no-any
+  processor: (i: any) => any = _ => _
```

Comment:
perhaps `preprocessor` is more suitable as it preprocesses the json before decoding it

```suggestion
  preprocessor: (i: any) => any = _ => _
```